### PR TITLE
Fix optional raven dependency not being optional [1.0.1]

### PR DESCRIPTION
**Pull request by justin-richert**
The main aim of this PR is to fix a bug in `extension.py` in which the code would blindly try to import parts of the `raven` package despite it not being a required dependency in the flask-logger PyPI package. The import now gracefully fails and informs the rest of the code execution as to whether or not `raven` was successfully imported. raven is also now listed as an optional dependency that can be automatically installed alongside flask-logger by doing `pip install flask-logger[raven]` or changing `flask-logger = "*"` to `flask-logger = {extras = ["raven"], version = "*"}` in a Pipfile.

Versioned to 1.0.1

Some other small things included in this PR:
- Updated README to reflect optional raven dependency and how to install with/without
- [fixed] pre-commit script was not included in the repo
- Now ignoring both .env and .envrc in .gitignore so users can pick between autoenv and direnv
- Disabled duplicate-code warnings in pylint globally; tried to do it at the file level and it refused to suppress it (research online seemed to suggest there is a pylint bug that other users are suffering from as well)
- Refactored Pipfile to correctly import raven and its optional Flask dependency on the same line (as shown in the lock file, the same packages are installed by either approach)
- Cleaned unneeded packages out of the Pipfile.lock caused by pipenv running in python2 at some point

### CODE REVIEW

File: flask_logger/tests/test_extension.py [hunk 1]
```diff
@@ -1,16 +1,19 @@
...
 
 from flask import Flask
 
 from flask_logger import Logger  # isort:skip
-from flask_logger.extension import LOGGERS  # isort:skip
+from flask_logger import extension  # isort:skip
 
 TEST_DSN = 'http://foo:bar@sentry.local/1?timeout=1'  # Got this from raven-python repo tests
 
+extension = importlib.reload(extension)
```
Comment (bbelyeu) [inline flask_logger/tests/test_extension.py]: Hmm, why is this necessary?

Comment (justin-richert) [reply flask_logger/tests/test_extension.py]: It actually isn't I don't think. I was getting test failures in this file saying that Client wasn't defined because I was reloading the module in the other test file. Both test modules get loaded in at the beginning before any execution, so one was breaking the other. I added this line to fix it, but I believe I actually fixed it by changing the import from `from flask_logger.extension import LOGGERS` to `from flask_logger import extension`. I believe that alone fixes it by actually having a reference to the module so it doesn't get refreshed when the other test file reloads it. All of that long-windedness to say, I think you're right - I'll remove this line.

Comment (justin-richert) [reply flask_logger/tests/test_extension.py]: Done.

File: setup.py [hunk 2]
```diff
@@ -18,6 +18,9 @@
     install_requires=[
         'Flask'
     ],
+    extras_require={
+        'raven': ['raven']
```
Comment (justin-richert) [inline setup.py]: One small nuance that I couldn't decide on:
Do you like the way this is named?

From the docs:
> A dictionary mapping names of “extras” (optional features of your project) to strings or lists of strings specifying what other distributions must be installed to support those features.

The optional feature is Sentry, but raven is the optional package. Maybe this should be
`extras_require={'Sentry': ['raven']}`? Probably doesn't matter all that much.

Comment (justin-richert) [reply setup.py]: Right now, installing the optional dependency would be like `pip install flask-logger[raven]`, but if I were to update it according to my comment above, it would be `pip install flask-logger[Sentry]`.

Comment (bbelyeu) [reply setup.py]: Ah... the optional feature is Sentry, but they named their package raven which is weird. I kind of like [Sentry] a little better. Any thoughts on why that wouldn't be the best way to go?

File: flask_logger/tests/test_extension.py (hunk 1 above)
Comment (bbelyeu) [reply flask_logger/tests/test_extension.py]: 👍

File: setup.py (hunk 2 above)
Comment (justin-richert) [reply setup.py]: Changed to the flask-logger[Sentry] option
